### DISCUSSION TITLE
Timestamp Missing for Lyrics Obtained via Subsonic API

### DISCUSSION THREAD

**Discussion by 8kdesign** [navidrome/navidrome#2982] [Q&A]
Hi! I noticed that lyrics obtained via getLyrics in Subsonic API would have its timestamp removed, which leads to the problem where I am unable to show synced lyrics in my app. This is inconsistent with Subsonic's implementation, which returns the lyrics without removing timestamp.

I noticed that it was done intentionally in this PR: https://github.com/navidrome/navidrome/issues/249

May I ask what is the reason for the decision to remove the timestamp? It doesn't really make sense to me.

**Comment by deluan** [comment]
The reason timestamps were removed is that most (all?) Subsonic clients do not expect timestamps when calling the `getLyrics` endpoint.

For that specific reason, a new endpoint, `getLyricsBySongId`, was [discussed](https://github.com/opensubsonic/open-subsonic-api/discussions/60) and [implemented](https://opensubsonic.netlify.app/docs/endpoints/getlyricsbysongid/) in OpenSubsonic.

**Reply by 8kdesign** [reply]
Hi! Thanks for the reply. It seems like Subsonic servers do not have the same behaviour though. When I run Subsonic from my Mac, the API returns the lyrics with timestamps.

**Reply by deluan** [reply]
Yes, the original [API documentation](https://www.subsonic.org/pages/api.jsp) can be very ambiguous and does not always match the behaviour of the original Subsonic server. Most of the behaviour implemented in Navidrome is based on my observations from the interaction between Subsonic and some old clients like DSub and Jamstash. As I'm a DSub user myself (transitioning to Symfonium), I noticed it was not properly showing lyrics with timestamps, so I decided to strip any timestamps from the `getLyrics` response.

If you are working on a new Subsonic client, I suggest making it compatible with [OpenSubsonic](https://github.com/opensubsonic), as it is what most, if not all, Subsonic-compatible servers are striving to implement and evolve together, with as much backward compatibility as possible.
